packages/core/src/external/term-server/index.ts

```diff
@@ -0,0 +1,112 @@
+import { TypedValue } from "@medplum/core";
+import { Coding, Parameters, ParametersParameter } from "@medplum/fhirtypes";
+import { createUuidFromText } from "@metriport/shared/common/uuid";
+import axios, { AxiosInstance } from "axios";
+import { Config } from "../../util/config";
+import {
+  CPT_URL,
+  CVX_URL,
+  ICD_10_URL,
+  LOINC_URL,
+  RXNORM_URL,
+  SNOMED_URL,
+} from "../../util/constants";
+import { EventMessageV1 } from "../analytics/posthog";
+
+export type CodeSystemLookupOutput = {
+  name: string;
+  id: string;
+  display: string;
+  code: string;
+  property?: { code: string; description: string; value: TypedValue }[];
+};
+
+export const termServerUrl = Config.getTermServerUrl();
+const bulkLookupUrl = "terminology/code-system/lookup/bulk";
+
+export const supportedSystems = [ICD_10_URL, SNOMED_URL, LOINC_URL, RXNORM_URL, CPT_URL, CVX_URL];
+
+export function isSystemValid(system: string) {
+  const trimmedSystem = system.trim();
+  if (supportedSystems.includes(trimmedSystem)) return true;
+  return false;
+}
+
+export function buildTermServerApi(): AxiosInstance | undefined {
+  if (!termServerUrl) return undefined;
+
+  return axios.create({
+    baseURL: termServerUrl,
+    timeout: 30_000,
```

Comment:
could prob set to something like 10s, since we're disregarding the request after 5s upstream anyways